### ISSUE TITLE
Setup CI

### ISSUE THREAD

**Issue by loewenstein** [istio-ecosystem/istio-installer#5]
We should automatically verify that the new installer can install known Istio versions. As the current CI of the Istio community is CircleCI, we should deliver this as CircleCI config.

Things to verify:
* The installation is indeed modular:
   - Install Istio Citadel + Control plane and verify a service A can reach a service B via Envoy data plane 
   - Add an Istio Ingress Gateway and verify we can reach service A from outside the cluster 
   - Add Istio telemetry and verify we get metrics from the service mesh
   - Add Istio Policy and verify that it works
   - Add Istio Egress Gateway and verify that it works
* The installation can be updated without service degradation:
  - Update all Istio components and verify that the service mesh stays functional 

I propose we verify the installation of Istio 1.1.0 and the update to the branch release-1.1.

**Comment by loewenstein** [comment]
@costinm I'd like to add job that verifies modular installation. Is there an e2e test that would work with Citadel and istio-control alone? Just looked into the [simple e2e test](https://github.com/istio/istio/tree/master/tests/e2e/tests/simple), but that already needs a Gateway.

**Comment by costinm** [comment]
There is a way to deploy Pilot + Gateway alone (in different NS) - this should work for simple.

**Comment by holgero** [comment]
The PR #21 has added a Circle CI job that tested the installation. It seemed to work from https://circleci.com/gh/istio-ecosystem/istio-installer/73 to https://circleci.com/gh/istio-ecosystem/istio-installer/122 (5 days ago), but is broken at the moment.

I think we should have some circle CI label in the readme, so that we have a visual clue for the CI job failing. What do you say, @costinm ?

**Comment by holgero** [comment]
Hi @costinm, we now have an installation/upgrade pipeline running on concourse https://concourse.istio.sapcloud.io/ where we could imagine to also run a PR validation. I think we need to grant members of the istio org viewer access to the build logs there before we can do that.
Concourse can understand github.com oauth and organisation membership, but one Istio org administrator would need to permit that workflow. Do you think that is feasible?

**Comment by holgero** [comment]
@costinm ?

**Comment by holgero** [comment]
@sdake @linsun @howardjohn ?

**Comment by costinm** [comment]
Hi - sorry for the delay, had a few other fires. 
Yes, I think it is great - we want to verify the installer works with multiple CI/CD systems.

@linsun  is admin on istio org, /lgtm

**Comment by costinm** [comment]
Is there a link we can add to the README ?

**Comment by costinm** [comment]
Also - I don't seem to have permission to the jobs (401)

**Comment by linsun** [comment]
I can't view either of these 2 links you posted earlier @holgero 

We are open to integrate with other CIs, it will be optional first till it is proved to stable then we can move to required.

A little tight on time before kubecon, but please specify exactly what admin support you need.

**Comment by holgero** [comment]
The earlier posted links in https://github.com/istio/installer/issues/5#issuecomment-484031906 no longer work for me too. They worked while the istio installer was in istio-ecosystem.
Here is the new location: https://circleci.com/gh/istio/istio-installer

The newer link  https://concourse.istio.sapcloud.io/ is the more interesting one. If you follow it, you will see a concourse UI with public pipelines displayed. In the top-right corner there is a login button which brings you to a page where you have the choice to login with Github. We configured the concourse to allow all github users that are members of the Istio org to be granted viewer role on the concourse. But to check if someone is a member of the Istio org, that org has to allow the oauth workflow. And that workflow needs someone with admin permission to allow it.

I just wanted to check if it is okay to ask for that permission. I can also make a screen shot if that helps.

**Comment by holgero** [comment]
But I just tried it myself and it didn't seem to work any more at the moment. And the developer who did the setup isn't here today. Perhaps we can give it a try next week, if your time permits? Otherwise we postpone this after the kubecon.

**Comment by costinm** [comment]
It is great to have this, and login with github seems good idea ( but the redirect is broken indeed).
We can wait - if we get it ready before the freeze on 21 it would be great. 

If I understand the script correctly, they run install and test on a k8s cluster ? Do you support using 
Kind ? I suppose when logs become visible I can answer my own q.

**Comment by phil9909** [comment]
@costinm I added your github user explicitly. You should now be able login to [our concourse](https://concourse.istio.sapcloud.io/) and view the logs.

**Comment by holgero** [comment]
I got the redirect page working for me again (had to remove the application from my github settings page to see it again). Now I got the chance to let it send a request to Istio org owners to access organization data. Org owners should see now that request.

**Comment by holgero** [comment]
Thanks to @geeknoid it works now. Members of the Istio org can now view the build logs on our concourse instance.
